src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -152,148 +137,140 @@ public void deserializeFromArrowBase64(String data) {
         // Clear the cache
         clearCache();
 
-        logger.debug("deserializeFromArrowBase64() returns");
+        logger.debug("deserializeQueryContextJson() returns");
         // Log existing cache entries
         logCacheEntries();
 
         return;
       }
 
-      byte[] decoded = Base64.getDecoder().decode(data);
-      ByteArrayInputStream input = new ByteArrayInputStream(decoded);
-      try (ArrowStreamReader reader = new ArrowStreamReader(input, new RootAllocator())) {
-        while (reader.loadNextBatch()) {
-          VectorSchemaRoot root = reader.getVectorSchemaRoot();
-          for (int i = 0; i < root.getRowCount(); ++i) {
-            QueryContextElement qce = deserializeEntry(root, i);
-            // Merge the element in the existing cache
-            merge(qce.id, qce.readTimestamp, qce.priority, qce.context);
+
+
+    try{
+      ObjectMapper mapper = new ObjectMapper();
+      JsonNode rootNode = mapper.readTree(data);
+
+      // Deserialize the main entry
+      JsonNode mainEntryNode = rootNode.path("main_entry");
```

Comment:
Can we put comment on what's main_entry v/s entries? A sample JSON with expected response will helpful.

Reply:
OK I can write a sample in the comment

Reply:
Lets just confirm that the main entry id is always 0 and the priority should be 0 as well so that it does not get kicked out of the cache.